src/js/net.js

```diff
@@ -403,15 +403,20 @@ function onread(socket, nread, isEOF, buffer) {
       return;
     }
 
-    var str = buffer.toString();
+    // We know for sure the last 6 characters are going to be the ending
+    // Lets create a buffer with those 6 characters without toString conversion.
+    var eofLength = 6;
+    var bufferLength = buffer.length;
+    var ending = new Buffer(eofLength);
+    buffer.copy(ending, 0, bufferLength - eofLength, bufferLength);
+    var expectedEnding = new Buffer('\\e\\n\\d');
     var eofNeeded = false;
-    if (str.length >= 6
-      && str.substr(str.length - 6, str.length) == '\\e\\n\\d') {
+    if (bufferLength >= 6 && ending.compare(expectedEnding) == 0) {
```

Comment:
Would it be feasible to read the last bytes from the buffer and compare their values? That way we would not need extra buffers.